### ISSUE TITLE
CodePush ReactNative "auto roll-back" iOS app version after reopen app

### ISSUE THREAD

**Issue by longb1997** [microsoft/react-native-code-push#2436]
on iOS, After modifying some features and stuff in our app we just released both new versions over the air and all goes ok. The app discovers the update accordingly, then installs it and so restarts the app.

Everything was ok until I realize that when I restart the app it kind of roll-back to the default store version (w/o our new updates).

But on android, it is working fine.

AppDelegate.mm
```
#import <CodePush/CodePush.h>

...

- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
#else
 // return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
  return [CodePush bundleURL];
#endif
}
```
### Environment

* react-native-code-push version: ^7.1.0
* react-native version: 0.70.6

**Comment by mzruiz** [comment]
Currently experiencing this as well. First time the app is downloaded we see the updates.

When you close the app and re-download it reverts back to the App Store version.

**Comment by vfmachado** [comment]
Same here!!

already commented in other similar issues, any updates?

**Comment by Greanie21** [comment]
For me it's happening both on android and IOS.
Any update?

**Comment by gshenar** [comment]
We are also running into a similar issue that started recently. Anyone find a fix yet?

**Comment by akinlekan28** [comment]
I am also experiencing this

**Comment by chauhanharsh3482** [comment]
I am experiencing the same, urgent fix needed please help

**Comment by choijiho0021** [comment]
project 1 not working ( IOS - auto rollback error ) 
project 2 working (android, ios -> not auto rollback error ) 
I'm confused. Why is it that something works and something doesn't?

**Comment by malikzype** [comment]
Facing the same issue on Android with CodePush 7.1.0 and RN 0.70.6. Can anyone help, please?

**Comment by JB-CHAUVIN** [comment]
Same issue goes for me...
- I start the app it's in version X1
- I reload my app, it's in version X2
- App is killed, starting it back, it's version X1.
- ...

**Comment by anhkieet** [comment]
For everyone who is facing this issue.

If you set the CodePush check manually. Don't forget to wrap your App component by HOC

![Screen Shot 2023-03-09 at 15 31 44](https://user-images.githubusercontent.com/11927500/223965279-986cd003-d1f0-489b-a338-331b132083bc.png)

**Comment by akinlekan28** [comment]
@anhkieet how do i edit in this case, i am using expo bare workflow
![Screenshot 2023-03-09 at 12 25 50 PM](https://user-images.githubusercontent.com/15618006/224022562-2fce2b01-0563-49bb-864b-886bbb3770d9.png)

**Comment by ducgiangtrankma** [comment]
Same issue

**Comment by ducgiangtrankma** [comment]
> on iOS, After modifying some features and stuff in our app we just released both new versions over the air and all goes ok. The app discovers the update accordingly, then installs it and so restarts the app.
> 
> Everything was ok until I realize that when I restart the app it kind of roll-back to the default store version (w/o our new updates).
> 
> But on android, it is working fine.
> 
> AppDelegate.mm
> 
> ```
> #import <CodePush/CodePush.h>
> 
> ...
> 
> - (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
> {
> #if DEBUG
>   return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
> #else
>  // return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
>   return [CodePush bundleURL];
> #endif
> }
> ```
> 
> ### Environment
> * react-native-code-push version: ^7.1.0
> * react-native version: 0.70.6

Has the problem been resolved yet?

**Comment by srinivastalnikar** [comment]
Experiencing the same. Any leads?

**Comment by vision-llz** [comment]
Is this problem solved?

**Comment by baixjian** [comment]
这个问题解决了吗？

**Comment by vision-llz** [comment]
> 这个问题解决了吗？

我的项目是被其他codepush插件影响到的，已经解决，你可以检查一下

**Comment by baixjian** [comment]
试了issues中很多办法都还是解决不了，只有注释掉原生回滚代码的逻辑了，想问一下您用的codepush的什么插件

**Comment by vision-llz** [comment]
> 试了issues中很多办法都还是解决不了，只有注释掉原生回滚代码的逻辑了，想问一下您用的codepush的什么插件

项目额外安装了expo-update，我卸载就好了

**Comment by alihamza1729** [comment]
Any solution for this problem?

**Comment by Phong-TH** [comment]
I am also experiencing this, iOS only.

**Comment by SravanthiGujjula** [comment]
Facing same issue in iOS.

**Comment by MikhailSuendukov** [comment]
Hi everyone!
Could you please confirm what this issue still persist on latest version (8.2.1)?

**Comment by LouisKraemer** [comment]
@MikhailSuendukov I can confirm the issue is still happening on version 8.2.1

**Comment by jibinbajuMTN** [comment]
same on my side

**Comment by rizshivalli** [comment]
@MikhailSuendukov  I can also confirm the issue is still happening on version 8.2.1

**Comment by ofito** [comment]
We had this problem for several months and recently we found the solution for our case. It seems that we had residual code after a react native upgrade:


https://react-native-community.github.io/upgrade-helper/?from=0.70.15&to=0.71.0#RnDiffApp-ios-RnDiffApp-AppDelegate.mm

Exactly in the `didFinishLaunchingWithOptions` inside of `AppDelegate.mm`, removing this part make it works:

```
RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];

NSDictionary *initProps = [self prepareInitialProps];	
  UIView *rootView = RCTAppSetupDefaultRootView(bridge, @"RnDiffApp", initProps);

if (@available(iOS 13.0, *)) {	
    rootView.backgroundColor = [UIColor systemBackgroundColor];	
  } else {	
    rootView.backgroundColor = [UIColor whiteColor];	
  }	
 	
  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];	
  UIViewController *rootViewController = [UIViewController new];	
  rootViewController.view = rootView;	
  self.window.rootViewController = rootViewController;	
  [self.window makeKeyAndVisible];	
  return YES;	
}
```

That was causing the `sourceURLForBridge` to run twice and therefore CodePush was initialised twice marking the update as failed and doing the rollback.

I hope you can find this information useful.
